### ISSUE TITLE
Custom validation messages set via buildForm

### ISSUE THREAD

**Issue by koichirose** [kristijanhusak/laravel-form-builder#225]
I have the following in my buildForm() function:
```php
$this
            ->add('title', 'text', [
                'label' => 'Title',
                'rules' => 'required'
            ])
```

Is there a way to set custom validation messages for the 'required' field directly in that array? I couldn't find anything.
Since this is needed for each form of that type, I'd like to avoid doing that in the controller as shown in the docs:
```php
$form->validate(['title' => 'required'], [
    'title.required' => 'Please provide a valid title.'
]);
```

**Comment by kristijanhusak** [comment]
It's not possible to do it that way at this moment, but it is a good idea. I'll mark it as feature and work on it when i get some time.

**Comment by koichirose** [comment]
Thank you. As always, let me know if you need help with testing it.

**Comment by kristijanhusak** [comment]
@koichirose I added this in the latest release (1.7.0). You can update and start using it.
There is short explanation in the docs here http://kristijanhusak.github.io/laravel-form-builder/overview/validation.html
If it works, please close the issue.

**Comment by koichirose** [comment]
@kristijanhusak it doesn't work for child forms.
Example (userform):
`
<?php namespace App\Forms;

use Kris\LaravelFormBuilder\Form;

class UserForm extends Form
{
    protected $clientValidationEnabled = false;
    
    public function buildForm()
    {
        $model = $this->getModel();
        if ($model) {
            $model->load('user_data');
        }

        $this
            ->add('name', 'text', [
                'rules' => 'required',
                'label' => trans('users.name'),
                'error_messages' => [
                    'name.required' => 'testing error message parent'
                ],
            ])
            [.....more fields.....]
        ->add('user_data', 'form', [
                'class' =>  'App\Forms\UserDataForm',
                'wrapper' => false,
                'model' => $this->getModel(),
                'label' => false,
            ])
`

UserDataForm:
`
<?php namespace App\Forms;

use Kris\LaravelFormBuilder\Form;

class UserDataForm extends Form
{
    protected $clientValidationEnabled = false;
    public function buildForm()
    {
        $model = $this->getModel();
        $model = ($model instanceof \App\Models\User) ? $model : null;

        $this
            ->add('first_name', 'text', [
                'rules' => 'required',
                'label' => trans('user_data.first_name'),
                'error_messages' => [
                    'first_name.required' => 'testing error message child'
                ],
            ])
`

I disabled client validation.
Leaving both fields empty shows this:
`
testing error message parent
The Name field is required. 
`

**Comment by kristijanhusak** [comment]
you need to namespace the error messages properly. I'm not managing that at all. so in your child form something like this should work:

```php
<?php namespace App\Forms;

use Kris\LaravelFormBuilder\Form;

class UserDataForm extends Form
{
    protected $clientValidationEnabled = false;
    public function buildForm()
    {
        $model = $this->getModel();
        $model = ($model instanceof \App\Models\User) ? $model : null;

        $this
            ->add('first_name', 'text', [
                'rules' => 'required',
                'label' => trans('user_data.first_name'),
                'error_messages' => [
                    'user_data.first_name.required' => 'testing error message child'
                ],
            ])
```

**Comment by kristijanhusak** [comment]
Let me know if that works. I'll probably set it up so it collects child form name and append it.

**Comment by koichirose** [comment]
It works by namespacing the error messsages.
I guess that by doing that it wouldn't work if the form is used as a parent and not a child.

Let me know if you manage to add it automatically if it's a child.

Thanks

**Comment by kristijanhusak** [comment]
@koichirose I fixed it in the latest version (1.7.10). Please update and let me know if it works.

**Comment by koichirose** [comment]
It works both for parent and child without namespacing.
Thank you!
